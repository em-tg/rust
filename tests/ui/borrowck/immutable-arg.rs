//@ check-pass
fn foo(_x: u32) {
    _x = 4;
    //~^ WARNING cannot assign to immutable argument `_x`
}

fn main() {}
