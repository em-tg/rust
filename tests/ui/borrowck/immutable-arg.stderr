warning[E0384]: cannot assign to immutable argument `_x`
  --> $DIR/immutable-arg.rs:3:5
   |
LL |     _x = 4;
   |     ^^^^^^ cannot assign to immutable argument
   |
   = note: `#[warn(mut_non_mut)]` on by default
help: consider making this binding mutable
   |
LL | fn foo(mut _x: u32) {
   |        +++

warning: 1 warning emitted

For more information about this error, try `rustc --explain E0384`.
